Control/Monad/Error/Class.hs

```diff
@@ -190,3 +194,21 @@ instance (Monoid w, MonadError e m) => MonadError e (LazyWriter.WriterT w m) whe
 instance (Monoid w, MonadError e m) => MonadError e (StrictWriter.WriterT w m) where
     throwError = lift . throwError
     catchError = StrictWriter.liftCatch catchError
+
+-- | MonadError analog to the 'try' function.
+tryError :: MonadError e m => m a -> m (Either e a)
+tryError action = (liftM Right action) `catchError` (return . Left)
+
+-- | Modify the value (but not the type) of an error.  The type is
```

Comment:
It might be worth mentioning the very similar `withExcept` (or `withExceptT`) function here.

Reply:
resolved